src/polygon/src/polygon.cpp

```diff
@@ -192,900 +165,486 @@ void PrintFailure(const std::exception &e, const PolygonsIdx &polys,
               << triangles[j][2] << std::endl;
   }
 }
+
+#define PRINT(msg) \
+  if (params.verbose) std::cout << msg << std::endl;
 #else
-#define OVERLAP_ASSERT(condition, msg) \
-  if (!(condition)) return true;
 #define PRINT(msg)
 #endif
 
 /**
- * The class first turns input polygons into monotone polygons, then
- * triangulates them using the above class.
+ * Ear-clipping triangulator based on David Eberly's approach from Geometric
+ * Tools, but adjusted to handle epsilon-valid polygons, and including a
+ * fallback that ensures a manifold triangulation even for overlapping polygons.
+ * This is an O(n^2) algorithm, but hopefully this is not a big problem as the
+ * number of edges in a given polygon is generally much less than the number of
+ * triangles in a mesh, and relatively few faces even need triangulation.
+ *
+ * The main adjustments for robustness involve clipping the sharpest ears first
+ * (a known technique to get higher triangle quality), and doing an exhaustive
+ * search to determine ear convexity exactly if the first geometric result is
+ * within precision.
  */
-class Monotones {
+
+class EarClip {
```

Comment:
This is a terrible diff - this file is basically a rewrite. Considering the 1,000 lines of added tests, this PR is really a net loss of ~500 lines of code.